orttraining/orttraining/training_api/optimizer.cc

```diff
@@ -215,15 +215,17 @@ void Optimizer::Initialize(const std::string& optim_path_or_bytes,
     ORT_THROW_IF_ERROR(optim_sess_->RegisterExecutionProvider(execution_provider));
   }
 
-  ORT_THROW_IF_ERROR(optim_sess_->Load(optim_path_or_bytes));
+  ORT_THROW_IF_ERROR(model_identifiers.optim_model.has_value() ? optim_sess_->Load(model_identifiers.optim_model.value()) : optim_sess_->Load(model_identifiers.optim_model_data, model_identifiers.optim_model_len));
+
   ORT_THROW_IF_ERROR(optim_sess_->Initialize());
 
   // Make sure that the checkpoint state can copy tensors
   state_->optimizer_checkpoint_state.optimizer_session_data_transfer_mgr = &optim_sess_->GetDataTransferManager();
 
   utils::GetGraphInputOutputNames(optim_sess_, input_names_, output_names_);
 
-  optimizer_algo_ptr_ = OptimizerAlorithmFactory::CreateInstance(optim_path_or_bytes, group_count_);
+  // TODO[Ashwini]: Fix this before merging this PR - Needs hardcoding to ADAMW optimizer or updating CreateInstance to take in a pointer to model data buffer.
```

Comment:
I think we should try to think of a better way to determine which optimizer is being used. Re-reading the onnx model, ort format model, or the data buffer may not be the best way IMO.

Reply:
Agree... You mentioned in the scrum that you were updating this as part of your checkpoint PR so I haven't touched this. Did not want to duplicate efforts. Let's sync offline on this.

Reply:
I will not be addressing this as part of the minimal build / checkpoint format PR. I can pick it up as a separate work item and address it in another PR.

Reply:
Current PR allows re-reading when .onnx file is supplied and defaults to adamw optimizer when .ort format model is supplied.